tests/browser/events/method-calls.js

```diff
@@ -308,4 +308,26 @@ export default function() {
 		fire( ractive.find( '#return_zero' ), 'click' );
 		t.ok( !preventedDefault && !stoppedPropagation );
 	});
+
+	test( 'Extend a context event', t => {
+		let customContext = false;
+
+		const ractive = new Ractive({
+			el: fixture,
+			template: `<button on-click='@this.fire("customContextEvent", { customContext: true }, 1)'>{{foo}}</button>`,
+			data: { foo: 0 },
+			on: {
+				customContextEvent: (context, value) => {
+					ractive.set({ foo: value });
+					if (context && context.customContext === true) {
+						customContext = true;
+					}
+				},
+			}
+		});
+
+		fire( ractive.find( 'button' ), 'click' );
+		t.equal( ractive.get( 'foo' ), 1 );
+		t.ok(customContext);
+	});
```

Comment:
Might also want to test how this works when the source is a proxy and the event is re-fired in a component, when the re-fire is from the template or from code, and propagation (fire from very bottom and second level, receive at top level via template `on-*` or code `on`).

The following is done in a rush and may not work, but you probably get the idea.

```
const Component = Ractive.extend({
  template: `
    <button on-click="myevent">Click Me!</button>
  `
});

const AnotherComponent = Ractive.extend({
  components: { Component },
  template: `
    <Component on-myevent="@this.fire('yetanotherevent', { custom: 'context' }, 'arg1')" />
    <Component on-myevent="@this.fire('yetanotherevent', {}, 'noContext')" />
    <Component on-myevent="anotherevent" />
  `,
  on: {
    anotherevent: function(context){
      this.fire('yetanotherevent', { custom: 'context' }, 'arg1');
      this.fire('yetanotherevent', {}, 'arg1');
    }
  }
})

const App = Ractive.extend({
  components: { AnotherComponent },
  template: `
    <AnotherComponent on-yetanotherevent="receiver" on-myevent="receiver" />
  `,
  on: {
    receiver: function(context){},
    '*.yetanotherevent': function(context){},
    '*.myevent' : function(context){},
  }
});
```